### ISSUE TITLE
Suggestion: whitelist for unused variables/parameters

### ISSUE THREAD

**Issue by flaviovs** [slevomat/coding-standard#1074]
It would be good if unused variables and parameters sniffs supported a whitelist of names, because PHPCS suppression is not granular enough to always flag only the variables that are unused intentionally. This is common in other languages (e.g. Python's `_`)

Consider the following example:

```php
function onSomething($ev, $foo) {
    echo $foo;
}
```

Now `$ev` will generate a _SlevomatCodingStandard.Functions.UnusedParameter.UnusedParameter_. To suppress the error one can do this:

```php
// phpcs:disable SlevomatCodingStandard.Functions.UnusedParameter
function onSomething($ev, $foo) {
    echo $foo;
}
```

Now a few months later somebody changes the code to:
```php
// phpcs:ignore SlevomatCodingStandard.Functions.UnusedParameter
function onSomething($ev, $foo) {
    echo 'event triggered: ' . $ev;
}
```

Now the sniff will never catch the unused parameter `$foo`, unless the programmer remember to remove the suppression (unlikely, in my experience).

Same for variables:

```php
[$h, $m, $s] = explode(':', $time_hms);
// (...many lines of code...)
echo $h;
// Unused variables $m and $s
```
Someone suppress the unused variable error:

```php
// phpcs:ignore SlevomatCodingStandard.Variables.UnusedVariable
[$h, $m, $s] = explode(':', $time_hms);
// (...many lines of code...)
echo $h;
// All good.
```

Now someone decides to implement it differently:

```php
// phpcs:ignore SlevomatCodingStandard.Variables.UnusedVariable
[$h, $m, $s] = explode(':', $time_hms);
// (...many lines of code...)
echo get_hms_hour($time_hms);
// $h, $m, and $s now unused forever.
```

This suggestion entails adding a property to the sniffs:

```xml
<!-- Note: ideally it should do regex matching -->
<property name="whitelist" type="array" value="/^unused_.+/" />
```

Then one could write code like this:

```php
function onSomething($unused_ev, $foo) {
    echo $foo;

    [$h, $unused_m, $unused_s] = explode(':', $time_hms);

    echo $h;
}
```

This not only would solve PHPCS lack of granularity, but would make the code self-documenting and much more readable, IMHO.

**Comment by srigi** [comment]
I would love to see some sort of pattern configuration implemented too. I like how [ESLint is handling](https://eslint.org/docs/latest/rules/no-unused-vars#argsignorepattern) this case.

Adding the `// phpcs:ignore` is really annoying, if we could simple name unused arg with `$_`.
